test/functional/email_confirmations_controller_test.rb

```diff
@@ -99,6 +99,44 @@ class EmailConfirmationsControllerTest < ActionController::TestCase
         assert page.has_content?("Multi-factor authentication")
```

Comment:
Can we add an assertion that checks that there's the otp or recovery heading (makes sure it does not render the  recovery code heading)?